src/main/java/com/iexec/worker/result/ResultService.java

```diff
@@ -443,4 +429,49 @@ public String computeResultDigest(ComputedFile computedFile) {
         }
         return resultDigest;
     }
+
+    // region Purge
+
+    /**
+     * Purge results from given task, especially its result folder.
+     *
+     * @param chainTaskId ID of the task to purge.
+     * @return {@literal true} if task folder has been deleted
+     * and task data has been cleared from this service;
+     * {@literal false} otherwise.
+     */
+    @Override
+    public boolean purgeTask(String chainTaskId) {
+        final String taskBaseDir = workerConfigService.getTaskBaseDir(chainTaskId);
+
+        resultInfoMap.remove(chainTaskId);
+        FileHelper.deleteFolder(taskBaseDir);
+
+        final boolean deletedInMap = !resultInfoMap.containsKey(chainTaskId);
+        final boolean deletedTaskFolder = !new File(taskBaseDir).exists();
+
+        boolean deleted = deletedInMap && deletedTaskFolder;
+        if (deletedTaskFolder) {
+            log.info("The result of the chainTaskId has been deleted [chainTaskId:{}]", chainTaskId);
+        } else {
+            log.warn("The result of the chainTaskId couldn't be deleted [chainTaskId:{}, deletedInMap:{}, " +
+                            "deletedTaskFolder:{}]",
+                    chainTaskId, deletedInMap, deletedTaskFolder);
+        }
+
+        return deleted;
```

Comment:
Maybe check if nothing remains after the purge ?
```suggestion
        return !existInMap && !folder.exist();
```

Reply:
That's what's actually done, isn't it?
https://github.com/iExecBlockchainComputing/iexec-worker/blob/5fe56e09aa37d17eed7baad07848d3ee93d26ccb/src/main/java/com/iexec/worker/result/ResultService.java#L448-L449

Reply:
I should have been more focused while reading, sorry